network/local.go

```diff
@@ -277,27 +277,47 @@ func (lc *LocalConn) Type() ConnType {
 // All operations are thread-safe.
 // The messages are marshalled and stored in the queue as a slice of bytes.
 type connQueue struct {
-	*sync.Cond
-	queue  [][]byte
-	closed bool
+	incomingQueue chan []byte
+	outgoingQueue chan []byte
+	closeCh       chan bool
+	wg            sync.WaitGroup
```

Comment:
The usage of this waiting group is a little confusing: There is never a "group of events" or a "collection of goroutines to finish" or to wait on; instead it is always just one event to wait for (closing). In other words: the group gets incremented exactly *once*, [here](https://github.com/dedis/cothority/pull/673/files#diff-758a279d81fc93cf992b979af141f431R296). To me it seems a channel would be the right tool to signal: "hey we are done here". https://golang.org/pkg/sync/#WaitGroup

This said, **I still approve of this pull** as it fixes many other issues (also this little detail is buried in unexported methods).